pkg/generate/app/sourcelookup.go

```diff
@@ -467,13 +470,17 @@ func StrategyAndSourceForRepository(repo *SourceRepository, image *ImageRef) (*B
 		source.DockerfileContents = repo.Info().Dockerfile.Contents()
 	}
 	if !repo.ignoreRepository {
-		remoteURL, err := repo.RemoteURL()
+		remoteURL, ok, err := repo.RemoteURL()
 		if err != nil {
 			return nil, nil, fmt.Errorf("cannot obtain remote URL for repository at %s", repo.location)
 		}
-		source.URL = remoteURL
-		source.Ref = remoteURL.Fragment
-		source.ContextDir = repo.ContextDir()
+		if ok {
+			source.URL = remoteURL
+			source.Ref = remoteURL.Fragment
+			source.ContextDir = repo.ContextDir()
+		} else {
+			source.Binary = true
```

Comment:
Fixed, ptal

On Tue, Mar 29, 2016 at 9:53 AM, Cesar Wong <notifications@github.com>
wrote:

> In pkg/generate/app/sourcelookup.go
> <https://github.com/openshift/origin/pull/8260#discussion_r57726132>:
>
> >  		if err != nil {
> >  			return nil, nil, fmt.Errorf("cannot obtain remote URL for repository at %s", repo.location)
> >  		}
> > -		source.URL = remoteURL
> > -		source.Ref = remoteURL.Fragment
> > -		source.ContextDir = repo.ContextDir()
> > +		if ok {
> > +			source.URL = remoteURL
> > +			source.Ref = remoteURL.Fragment
> > +			source.ContextDir = repo.ContextDir()
> > +		} else {
> > +			source.Binary = true
>
> Yup, missed this. I do agree. If I give you a context dir I would expect
> it to be honored even if I'm only going to do binary builds.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8260/files/e86406dd0583f043cd1463b00e56fe24aa8b7f49#r57726132>
>